### PR TITLE
feat(docs): update guide with script tool

### PR DESCRIPTION
**Pull request by RealShadowNova**
- Add `scripts/generate.mjs` script prompter tool.
- Change `docs/` to implement new generation script tool.

### CODE REVIEW

File: docs/ContributingYourOwnProvider.md [hunk 1]
```diff
@@ -1,38 +1,46 @@
-# Writing your provider for Josh
+# Contributing your own provider for Josh
 
-This page will guide you through the development process of creating your provider.
+This page will help guide you through the process of creating and contributing your own provider to the Josh database wrapper project. We are glad that you would like to contribute to our amazing project. Please follow these instructions carefully to help ensure your contributions are accepted.
```
Comment (dan-online) [inline docs/ContributingYourOwnProvider.md:3]: > creating or contributing to a provider

would work better here

File: docs/ContributingYourOwnProvider.md [hunk 2]
```diff
@@ -1,38 +1,46 @@
...
-You should have already read the previous [Getting Started](GettingStarted.md) page. If you haven't read it please go back and do so.
+To be able to contribute your own provider for this project you are required to have extensive knowledge with [TypeScript](https://www.typescriptlang.org). In order to get your workspace ready you will need to follow the steps in [`CONTRIBUTING.md`](.github/CONTRIBUTING.md).
 
-## Creating your provider
+## Getting Started
 
-First, before we create any files let's create a sub-directory in `src/` called `lib` using the following command.
+```sh
+# Initiate Yarn in this repository.
+yarn
```
Comment (dan-online) [inline docs/ContributingYourOwnProvider.md:13]: Specify that yarn is a requirement because we use workspaces

File: scripts/generate.mjs [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+import prompts from 'prompts';
+
+const response = await prompts([
+	{ type: 'text', name: 'name', message: "What's the name of your provider?", initial: 'mydb' },
+	{ type: 'text', name: 'title', message: "What's the title case name of your provider?", initial: 'MyDB' },
+	{ type: 'text', name: 'description', message: 'Write a short description of your provider', initial: 'A Josh provider' },
+	{ type: 'select', name: 'umd', message: 'Can your provider be used in a browser?', choices: ['Yes', 'No'] }
+]);
+
+if (response.name === undefined) process.exit(1);
```
Comment (dan-online) [inline scripts/generate.mjs:16]: Perhaps also log an error if this code has the possibility to occur

File: scripts/generate.mjs [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+
+	spinner.succeed();
+}
+
+async function generatePackageFile() {
+	const spinner = ora('Package File Generation').start();
+
+	await writeFile(
+		resolve(provider, 'package.json'),
+		JSON.stringify(
```
Comment (dan-online) [inline scripts/generate.mjs:53]: I believe this generates uglified code so perhaps pass the correct arguments to pad the output

File: scripts/generate.mjs [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+	});
+
+	spinner.succeed();
+}
+
+async function generateConfigurationFiles() {
+	const spinner = ora('Configuration File Generation').start();
+
+	await writeFile(
+		resolve(provider, 'jest.config.ts'),
```
Comment (dan-online) [inline scripts/generate.mjs:136]: Maybe move this to a file instead of this method but it's not too necessary

**Review by dan-online — CHANGES_REQUESTED** [review]
Just a couple changes but overall looks good, I like the nice touch of a spinner

File: scripts/generate.mjs (hunk 3 above)
Comment (RealShadowNova) [reply scripts/generate.mjs:16]: This only occurs when you `^C` during the prompts.

File: docs/ContributingYourOwnProvider.md [hunk 6]
```diff
@@ -1,38 +1,48 @@
-# Writing your provider for Josh
+# Contributing your own provider for Josh
 
-This page will guide you through the development process of creating your provider.
+This page will help guide you through the process of creating and/or contributing a provider or your own to the Josh database wrapper project. We are glad that you would like to contribute to our amazing project. Please follow these instructions carefully to help ensure your contributions are accepted.
```
Comment (dan-online) [inline docs/ContributingYourOwnProvider.md:3]: Repetitive "or" here

**Review by dan-online — APPROVED** [review]
LGTM

**Review by dan-online — APPROVED** [review]

**Pull request merged**